frontend/src/Pages/Events/EventCreate.svelte

```diff
@@ -1,26 +1,49 @@
 <script>
     import { navigate } from "svelte-routing";
 
+    export let idOfEventToEdit = undefined;
     async function getFormData() {
-        let eventTypeReponse = await fetch('/api/eventtypes/');
-        let groupResponse = await fetch('/api/groups/');
-        let officerResponse = await fetch('/api/officers/');
-
-        let eventTypes = await eventTypeReponse.json();
-        let groups = await groupResponse.json();
-        let officers = await officerResponse.json();
-
-        return {
-            "eventTypes": eventTypes,
-            "groups": groups,
-            "officers": officers
+        let getJSON = (response) => response.json();
+        let [eventTypes, groups, officers] = await Promise.all([
+            fetch("/api/eventtypes/").then(getJSON),
+            fetch("/api/groups/").then(getJSON),
+            fetch("/api/officers/").then(getJSON),
+        ]);
+
+        let formData = {
+            eventTypes: eventTypes,
+            groups: groups,
+            officers: officers,
+        };
+
+        if (idOfEventToEdit != undefined) {
+            formData["eventToEdit"] = await fetch(
+                `/api/events/${idOfEventToEdit}/`
+            ).then(getJSON);
+            // awful way to make dates fit; what if the date format changes?
+            // TODO: find a better method to make API data fit form, perhaps
+            //  by simply only storing minutes or somthing.
+            let str = formData["eventToEdit"].start_time;
```

Comment:
Looks like timezones are not being accounted for with this either. When editing an event it seems to convert the time to GMT and then when you save, the time of the event moves by 7 hours. Maybe look into the javascript date object to see how you can make this conversion stronger